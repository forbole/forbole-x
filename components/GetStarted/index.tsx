import { Box, Button, Typography, useTheme } from '@material-ui/core'
import React from 'react'
import useTranslation from 'next-translate/useTranslation'
import GetStartedLightImage from '../../assets/images/get_started_light.svg'
import GetStartedDarkImage from '../../assets/images/get_started_dark.svg'
import useStyles from './styles'
import { useGeneralContext } from '../../contexts/GeneralContext'
import OnboardingDialog from '../OnboardingDialog'
import { useWalletsContext } from '../../contexts/WalletsContext'
import CreateWalletDialog from '../CreateWalletDialog'
import useIsMobile from '../../misc/useIsMobile'

const GetStarted: React.FC = () => {
  const classes = useStyles()
  const { theme } = useGeneralContext()
  const themeStyle = useTheme()
  const isMobile = useIsMobile()
  const { unlockWallets } = useWalletsContext()
  const { t } = useTranslation('common')
  const [isOnboardingDialogOpen, setIsOnboardingDialogOpen] = React.useState(false)
  const [isCreateWalletDialogOpen, setIsCreateWalletDialogOpen] = React.useState(false)

  return (
    <>
      <Box
        className={classes.container}
        mt={isMobile ? '5vh' : '15vh'}
        width={isMobile ? themeStyle.spacing(40) : 'inherits'}
      >
        {theme === 'light' ? (
          <GetStartedLightImage
            width={isMobile ? themeStyle.spacing(40) : themeStyle.spacing(60)}
            height={isMobile ? themeStyle.spacing(24.12) : 'inherits'}
          />
        ) : (
          <GetStartedDarkImage
            width={isMobile ? themeStyle.spacing(40) : themeStyle.spacing(60)}
            height={isMobile ? themeStyle.spacing(24.12) : 'inherits'}
          />
        )}
        <Typography
          className={classes.marginTop}
          variant={isMobile ? 'subtitle1' : 'h4'}
          align="center"
        >
          {t('get started line1')}
        </Typography>
        <Typography
          className={classes.marginTopSub}
          variant={isMobile ? 'subtitle1' : 'h6'}
          align="center"
        >
          {t('get started line2')}
        </Typography>
        <Button
          className={classes.button}
          variant="contained"
          color="primary"
          onClick={() => setIsOnboardingDialogOpen(true)}
        >
          {t('get started button')}
        </Button>
      </Box>
      <OnboardingDialog
        open={isOnboardingDialogOpen}
        onClose={() => setIsOnboardingDialogOpen(false)}
        onSubmit={(password) => {
          unlockWallets(password)
          setIsOnboardingDialogOpen(false)
          setIsCreateWalletDialogOpen(true)
        }}
      />
      <CreateWalletDialog
        open={isCreateWalletDialogOpen}
        onClose={() => setIsCreateWalletDialogOpen(false)}
      />
    </>
  )
}

export default GetStarted
