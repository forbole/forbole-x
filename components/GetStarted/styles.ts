import { makeStyles } from '@material-ui/core/styles'

const useStyles = makeStyles(
  (theme) => ({
    container: {
      alignSelf: 'center',
      display: 'flex',
      flexDirection: 'column',
      alignItems: 'center',
      marginTop: '15vh',
      width: theme.spacing(60),
    },
    marginTop: {
      marginTop: theme.spacing(3),
    },
    button: {
      marginTop: theme.spacing(3),
      width: theme.spacing(30),
    },
  }),
  {
    name: 'HookGlobalStyles',
    index: 2,
  }
)

export default useStyles
