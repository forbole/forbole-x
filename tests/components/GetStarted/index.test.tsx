import React from 'react';
import renderer from 'react-test-renderer';
import GetStarted from '../../../components/GetStarted';

const mockGeneralContext = {
  theme: 'light',
  setTheme: jest.fn(),
};

const mockWalletsContext = {
  unlockWallets: jest.fn(),
};

jest.mock('../../../contexts/GeneralContext', () => ({
  useGeneralContext: () => mockGeneralContext,
}));

jest.mock('../../../contexts/WalletsContext', () => ({
  useWalletsContext: () => mockWalletsContext,
}));

jest.mock('../../../components/OnboardingDialog', () => 'OnboardingDialog');
jest.mock('../../../components/CreateWalletDialog', () => 'CreateWalletDialog');

describe('component: GetStarted', () => {
  it('renders light theme correctly', () => {
    const component = renderer.create(<GetStarted />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders dark theme correctly', () => {
    mockGeneralContext.theme = 'dark';
    const component = renderer.create(<GetStarted />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('opens Onboarding Dialog on click', () => {
    const component = renderer.create(<GetStarted />);
    renderer.act(() => {
      component.root.findByType('button').props.onClick();
    });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('closes OnboardingDialog when onClose is called from Dialog', () => {
    const component = renderer.create(<GetStarted />);
    renderer.act(() => {
      component.root.findByType('button').props.onClick();
    });
    renderer.act(() => {
      component.root
        .findByProps({
          open: true,
        })
        .props.onClose();
    });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('opens CreateWalletDialog and calls setPassword when onSubmit is called from OnboardingDialog', () => {
    const component = renderer.create(<GetStarted />);
    renderer.act(() => {
      component.root.findByType('button').props.onClick();
    });
    renderer.act(() => {
      component.root
        .findByProps({
          open: true,
        })
        .props.onSubmit('123123');
    });
    expect(mockWalletsContext.unlockWallets).toBeCalledWith('123123');
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('closes CreateWalletDialog when onClose is called from CreateWalletDialog', () => {
    const component = renderer.create(<GetStarted />);
    renderer.act(() => {
      component.root.findByType('button').props.onClick();
    });
    renderer.act(() => {
      component.root
        .findByProps({
          open: true,
        })
        .props.onSubmit('123123');
    });
    renderer.act(() => {
      component.root
        .findByProps({
          open: true,
        })
        .props.onClose();
    });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

afterEach(() => {
  jest.clearAllMocks();
});
