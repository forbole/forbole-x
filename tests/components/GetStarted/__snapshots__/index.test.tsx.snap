// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: GetStarted closes CreateWalletDialog when onClose is called from CreateWalletDialog 1`] = `
Array [
  <div
    className="MuiBox-root MuiBox-root-9 HookGlobalStyles-container-1"
  >
    <svg
      data-file-name="SvgGet_started_dark"
    />
    <h4
      className="MuiTypography-root HookGlobalStyles-marginTop-2 MuiTypography-h4 MuiTypography-alignCenter"
    >
      common:get started line1
    </h4>
    <h6
      className="MuiTypography-root HookGlobalStyles-marginTop-2 MuiTypography-h6 MuiTypography-alignCenter"
    >
      common:get started line2
    </h6>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained HookGlobalStyles-button-3 MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        common:get started button
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>,
  <OnboardingDialog
    onClose={[Function]}
    onSubmit={[Function]}
    open={false}
  />,
  <CreateWalletDialog
    onClose={[Function]}
    open={false}
  />,
]
`;

exports[`component: GetStarted closes OnboardingDialog when onClose is called from Dialog 1`] = `
Array [
  <div
    className="MuiBox-root MuiBox-root-7 HookGlobalStyles-container-1"
  >
    <svg
      data-file-name="SvgGet_started_dark"
    />
    <h4
      className="MuiTypography-root HookGlobalStyles-marginTop-2 MuiTypography-h4 MuiTypography-alignCenter"
    >
      common:get started line1
    </h4>
    <h6
      className="MuiTypography-root HookGlobalStyles-marginTop-2 MuiTypography-h6 MuiTypography-alignCenter"
    >
      common:get started line2
    </h6>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained HookGlobalStyles-button-3 MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        common:get started button
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>,
  <OnboardingDialog
    onClose={[Function]}
    onSubmit={[Function]}
    open={false}
  />,
  <CreateWalletDialog
    onClose={[Function]}
    open={false}
  />,
]
`;

exports[`component: GetStarted opens CreateWalletDialog and calls setPassword when onSubmit is called from OnboardingDialog 1`] = `
Array [
  <div
    className="MuiBox-root MuiBox-root-8 HookGlobalStyles-container-1"
  >
    <svg
      data-file-name="SvgGet_started_dark"
    />
    <h4
      className="MuiTypography-root HookGlobalStyles-marginTop-2 MuiTypography-h4 MuiTypography-alignCenter"
    >
      common:get started line1
    </h4>
    <h6
      className="MuiTypography-root HookGlobalStyles-marginTop-2 MuiTypography-h6 MuiTypography-alignCenter"
    >
      common:get started line2
    </h6>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained HookGlobalStyles-button-3 MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        common:get started button
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>,
  <OnboardingDialog
    onClose={[Function]}
    onSubmit={[Function]}
    open={false}
  />,
  <CreateWalletDialog
    onClose={[Function]}
    open={true}
  />,
]
`;

exports[`component: GetStarted opens Onboarding Dialog on click 1`] = `
Array [
  <div
    className="MuiBox-root MuiBox-root-6 HookGlobalStyles-container-1"
  >
    <svg
      data-file-name="SvgGet_started_dark"
    />
    <h4
      className="MuiTypography-root HookGlobalStyles-marginTop-2 MuiTypography-h4 MuiTypography-alignCenter"
    >
      common:get started line1
    </h4>
    <h6
      className="MuiTypography-root HookGlobalStyles-marginTop-2 MuiTypography-h6 MuiTypography-alignCenter"
    >
      common:get started line2
    </h6>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained HookGlobalStyles-button-3 MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        common:get started button
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>,
  <OnboardingDialog
    onClose={[Function]}
    onSubmit={[Function]}
    open={true}
  />,
  <CreateWalletDialog
    onClose={[Function]}
    open={false}
  />,
]
`;

exports[`component: GetStarted renders dark theme correctly 1`] = `
Array [
  <div
    className="MuiBox-root MuiBox-root-5 HookGlobalStyles-container-1"
  >
    <svg
      data-file-name="SvgGet_started_dark"
    />
    <h4
      className="MuiTypography-root HookGlobalStyles-marginTop-2 MuiTypography-h4 MuiTypography-alignCenter"
    >
      common:get started line1
    </h4>
    <h6
      className="MuiTypography-root HookGlobalStyles-marginTop-2 MuiTypography-h6 MuiTypography-alignCenter"
    >
      common:get started line2
    </h6>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained HookGlobalStyles-button-3 MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        common:get started button
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>,
  <OnboardingDialog
    onClose={[Function]}
    onSubmit={[Function]}
    open={false}
  />,
  <CreateWalletDialog
    onClose={[Function]}
    open={false}
  />,
]
`;

exports[`component: GetStarted renders light theme correctly 1`] = `
Array [
  <div
    className="MuiBox-root MuiBox-root-4 HookGlobalStyles-container-1"
  >
    <svg
      data-file-name="SvgGet_started_light"
    />
    <h4
      className="MuiTypography-root HookGlobalStyles-marginTop-2 MuiTypography-h4 MuiTypography-alignCenter"
    >
      common:get started line1
    </h4>
    <h6
      className="MuiTypography-root HookGlobalStyles-marginTop-2 MuiTypography-h6 MuiTypography-alignCenter"
    >
      common:get started line2
    </h6>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained HookGlobalStyles-button-3 MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        common:get started button
      </span>
    </button>
  </div>,
  <OnboardingDialog
    onClose={[Function]}
    onSubmit={[Function]}
    open={false}
  />,
  <CreateWalletDialog
    onClose={[Function]}
    open={false}
  />,
]
`;
